### PR TITLE
Feature/calculate order amount

### PR DESCRIPTION
**Pull request by 123yael**
After you approve the two functions, I will update from main and only after that I will merge to main, thanks!

### CODE REVIEW

File: src/main/java/com/sapred/ordermanagerred/model/Currency.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.sapred.ordermanagerred.model;
+
+public enum Currency {
+    EURO(89.52907), SHEKEL(22.70258), FRANC(91.45229), RUBLE(0.97772), DOLLAR(81.98195);
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/model/Currency.java:4]: The enum value should be the short name of the currency not the exchange rate (the rate can be changed at any particular time so we should get it from an external web service)
EURO: EUR
SHEKEL: ILS
FRANC: CHF
RUBLE: RUB
DOLLAR: USD


https://www.eurochange.co.uk/travel/tips/world-currency-abbreviations-symbols-and-codes-travel-money

File: src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java [hunk 2]
```diff
@@ -26,6 +26,23 @@ public class ProductCategoryService {
...
+    public ResponseEntity<String> createProductCategory(ProductCategory productCategory){
+        try {
+            if(productCategoryRepository.existsByName(productCategory.getName()))
+                throw new ResponseStatusException(HttpStatus.CONFLICT,"the name of the category already exists");
+            AuditData auditData=new AuditData();
+            auditData.setCreateDate(new Date());
+            auditData.setUpdateDate(new Date());
+            productCategory.setAuditData(auditData);
+            productCategoryRepository.save(productCategory);
+            return ResponseEntity.ok("success: true");
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java:39]: ResponseEntity only in controller

File: src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java [hunk 3]
```diff
@@ -26,6 +26,23 @@ public class ProductCategoryService {
     @Autowired
     private ProductCategoryMapper productCategoryMapper;
 
+    //יצירת מופע של productCategory
+    public ResponseEntity<String> createProductCategory(ProductCategory productCategory){
+        try {
+            if(productCategoryRepository.existsByName(productCategory.getName()))
+                throw new ResponseStatusException(HttpStatus.CONFLICT,"the name of the category already exists");
+            AuditData auditData=new AuditData();
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java:34]: Please create new object by lomob SetupBuilder

File: src/main/java/com/sapred/ordermanagerred/service/OrderService.java [hunk 4]
```diff
@@ -53,5 +67,41 @@ else if (i % 3 == 1)
...
+                    companyRepository.findById("1").get(), new AuditData(new Date(), new Date()));
+            productCategoryRepository.save(pc);
+            Product p = new Product(i + "", "aaa", "aaa", 40, 50,
+                    DiscountType.PERCENTAGE, pc, 4,
+                    companyRepository.findById("1").get(), new AuditData(new Date(), new Date()));
+            productRepository.save(p);
+        }
+    }
+
+    public List<ProductCartDTO> calculateOrderAmount(Order order) {
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/service/OrderService.java:85]: Code is nice

The problem is that once the user press "Add" the system performs a server call to calculate the amount for the newly added product and in your code, you calculate the amount for every ordered item which already calculated in the previous "Add"
So maybe you can calculate the order item only for item which amount is empty and not for all the items

**Review by moriyahazan — CHANGES_REQUESTED** [review]

File: src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java (hunk 2 above)
Comment (123yael) [reply src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java:39]: It's not related to my mission, but I think it's fixed

File: src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java (hunk 3 above)
Comment (123yael) [reply src/main/java/com/sapred/ordermanagerred/service/ProductCategoryService.java:34]: It's not related to my mission, but I think it's fixed

**Review by moriyahazan — APPROVED** [review]

**Pull request merged**